### PR TITLE
Add HMR support

### PR DESCRIPTION
**Pull request by stevenkampen**
@andrewthauer This adds HMR support. It works in my local HMR configured app and on stackblitz.com (e.g. https://angular-9b55kc.stackblitz.io/).

### CODE REVIEW

File: src/utils/serialize.ts [hunk 1]
```diff
@@ -18,5 +18,5 @@ export const serialize = value => {
 
 /// Deserialize a function string and invoke the resulting object recreator.
 export const deserialize = value => {
-  return parse(value);
+  return value === 'null' || value === null ? null : parse(value);
```
Comment (andrewthauer) [inline src/utils/serialize.ts]: Curious if this was related to HMR or does it potentially resolve some of the reported regressions?

File: src/backend/backend.ts [hunk 2]
```diff
@@ -214,31 +205,20 @@ const updateRouterTree = (): Promise<void> => {
...
-    subject.debounceTime(0).subscribe(() => {
-      Promise.all([
-        updateComponentTree(getAllAngularRootElements().map(r => ng.probe(r))),
-        updateRouterTree()
-      ]).then(() => onUpdateNotifier.next());
-    }));
-
-  // initial load
-  subject.next(void 0);
+const listenForSomeTimeAndMaybeResubscribe = (timeMs: number) => {
```
Comment (andrewthauer) [inline src/backend/backend.ts]: Any chance something like this approach might work in `augury-labs` lazy loaded module support?

File: src/utils/serialize.ts (hunk 1 above)
Comment (stevenkampen) [reply src/utils/serialize.ts]: @andrewthauer This just avoids passing the `"null"` string into `parse` which throws an error. This only occurs (the `value` of `"null"`) when some error has happened in the backend (I think  with the serialization).

**Comment by andrewthauer** [comment]
@stevenkampen - Did you want to do a canary release of this fix and see which issues it fixes?

File: src/backend/backend.ts (hunk 2 above)
Comment (stevenkampen) [reply src/backend/backend.ts]: @andrewthauer Possibly, but it's all still a little hazy for now.

**Pull request merged**